src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+  @tailrec
+  private[ledger] final def executePreparedTransactions(
+    signedTransactions: Seq[SignedTransaction], world: InMemoryWorldStateProxy,
+    blockHeader: BlockHeader, signedTransactionValidator: SignedTransactionValidator,
+    acumGas: BigInt = 0, acumReceipts: Seq[Receipt] = Nil, executed: Seq[SignedTransaction] = Nil)
+  : Either[TxsExecutionError, (BlockResult, Seq[SignedTransaction])] = {
+
+    val result = executeTransactions(signedTransactions, world, blockHeader, signedTransactionValidator, acumGas, acumReceipts)
+
+    result match {
+      case Left(TxsExecutionError(stx, StateBeforeFailure(worldState, gas, receipts), reason)) if signedTransactions.length > 1 =>
```

Comment:
Also, in relation to what @AlanVerbner said, in which scenario would we want `executePreparedTransactions` to return something of type `Left`?
If the execution of all the pased txs failed, maybe we could return something of type `(BlockResult, Seq[SignedTransaction])` with the sequence of txs empty?